src/gtk/gftp-gtk.c

```diff
@@ -504,66 +509,81 @@ CreateConnectToolbar (GtkWidget * parent)
 
   gtk_box_pack_start (GTK_BOX (box), tempwid, FALSE, FALSE, 0);
 
-  hostedit = gtk_combo_new ();
-  gtk_combo_set_case_sensitive (GTK_COMBO (hostedit), 1);
-  gtk_widget_set_size_request (hostedit, 130, -1);
-
-  gtk_signal_connect (GTK_OBJECT (GTK_COMBO (hostedit)->entry), "activate",
-		      GTK_SIGNAL_FUNC (toolbar_hostedit), NULL);
+  hostedit = gtk_combo_box_text_new_with_entry ();
+  gtk_widget_set_size_request (hostedit, 120, -1);
```

Comment:
This is fine for this pull request since I know there's other places where the pixel counts are hard coded as well but something to keep in mind is HiDPI support. Try running gftp with the GDK_SCALE=2 environment variable set and the window will double in size.

Reply:
I noticed that GDK_SCALE=2 only works with gtk3. gftp currently only compiles with gtk2. So yeah that should be fixed.

I found this:
https://github.com/sedwards/gftp/commits/gtk3-rewrite

From what I can see everything is broken, it's just preliminary work. But there are some code lines that can be used to convert GtkCList code to GtkTreeView.

I try to make gradual changes, so that someone else can find a working app and continue making changes, when the code compiles with gtk 2.24 without deprecations, it will be ready to be ported to gtk3..